File: app/src/main/java/mb00/android/codehub/ui/repo/adapter/CommitAdapter.java
package mb00.android.codehub.ui.repo.adapter;

import android.content.Context;
import android.support.v4.content.ContextCompat;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import java.util.List;

import mb00.android.codehub.R;
import mb00.android.codehub.api.model.Commit;
import mb00.android.codehub.logic.utils.DateParser;
import mb00.android.codehub.ui.repo.view.RepoCommitsFragment;

/**
 * A RecyclerView adapter used to display commits in {@link RepoCommitsFragment}
 */

public class CommitAdapter extends RecyclerView.Adapter<CommitAdapter.CommitHolder> {

    private List<Commit> commitList;
    private Context context;

    public CommitAdapter(List<Commit> commitList, Context context) {
        this.commitList = commitList;
        this.context = context;
    }

    public class CommitHolder extends RecyclerView.ViewHolder {

        private ImageView commitAvatarImageView;
        private TextView commitTitleTextView;
        private TextView commitDetailTextView;

        public CommitHolder(View itemView) {
            super(itemView);

            commitAvatarImageView = itemView.findViewById(R.id.repo_commit_avatar_image_view);
            commitTitleTextView = itemView.findViewById(R.id.repo_commit_title_text_view);
            commitDetailTextView = itemView.findViewById(R.id.repo_commit_detail_text_view);
        }
    }

    @Override
    public CommitHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View commitView = inflater.inflate(R.layout.view_holder_commit, parent, false);
        return new CommitHolder(commitView);
    }

    @Override
    public void onBindViewHolder(CommitHolder holder, int position) {
        Commit commit = commitList.get(position);
        holder.commitTitleTextView.setText(commit.getCommitNested().getMessage());
        String details = commit.getCommitNested().getAuthor().getName() + " " + DateParser.parseEnglish(commit.getCommitNested().getAuthor().getDate());
        holder.commitDetailTextView.setText(details);

        try {
            Glide.with(context).load(commit.getAuthor().getAvatarUrl()).into(holder.commitAvatarImageView);
        } catch (NullPointerException npe) {
            Context context = holder.commitAvatarImageView.getContext();
            holder.commitAvatarImageView.setImageDrawable(ContextCompat.getDrawable(context, R.drawable.ic_github));
        }
    }

    @Override
    public int getItemCount() {
        return commitList.size();
    }

}

File: app/src/main/java/mb00/android/codehub/ui/repo/adapter/CommitAdapter.kt
package mb00.android.codehub.ui.repo.adapter

import android.content.Context
import android.support.v4.content.ContextCompat
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

import com.bumptech.glide.Glide

import mb00.android.codehub.R
import mb00.android.codehub.api.model.Commit
import mb00.android.codehub.logic.utils.DateParser
import mb00.android.codehub.ui.repo.view.RepoCommitsFragment

/**
 * A RecyclerView adapter used to display commits in [RepoCommitsFragment]
 */

class CommitAdapter(
        private val commitList: List<Commit>,
        private val context: Context
) : RecyclerView.Adapter<CommitAdapter.CommitHolder>() {

    inner class CommitHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val commitAvatarImageView: ImageView = itemView.findViewById(R.id.repo_commit_avatar_image_view)
        val commitTitleTextView: TextView = itemView.findViewById(R.id.repo_commit_title_text_view)
        val commitDetailTextView: TextView = itemView.findViewById(R.id.repo_commit_detail_text_view)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CommitHolder {
        val inflater = LayoutInflater.from(parent.context)
        val commitView = inflater.inflate(R.layout.view_holder_commit, parent, false)
        return CommitHolder(commitView)
    }

    override fun onBindViewHolder(holder: CommitHolder, position: Int) {
        val commit = commitList[position]
        holder.commitTitleTextView.text = commit.commitNested?.message
        val details = commit.commitNested?.author?.name + " " + DateParser.parseEnglish(commit.commitNested?.author?.date)
        holder.commitDetailTextView.text = details

        /*if (commit.author.avatarUrl.isNotEmpty())
            Glide.with(context).load(commit.author.avatarUrl).into(holder.commitAvatarImageView)
        else
            holder.commitAvatarImageView.setImageDrawable(ContextCompat.getDrawable(context, R.drawable.ic_github))*/

        commit.author.let {
            Glide.with(context).load(it?.avatarUrl).into(holder.commitAvatarImageView)
        }
    }

    override fun getItemCount(): Int {
        return commitList.size
    }

}
